PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -83,8 +84,26 @@ function transformReports(inputReportFiles: string[], index: number)
         sarifReports.push(report);
         processResults(inputReportFiles, index);
     } else if (report.toLocaleLowerCase().endsWith(XML_EXTENSION)) {
+        ruleAnalyzerPairs.clear();
+
         const saxStream = sax.createStream(true, {});
         saxStream.on("opentag", function (node) {
+            if (node.name == 'Rule' && !bLegacyReport) {
+                ruleAnalyzerPairs.set(node.attributes.id, node.attributes.analyzer);
+            }
+
+            if (bLegacyReport) {
```

Comment:
You may want to check whether rule already exists in the map here to do earlier condition check.